public/app/features/dashboard/dashgrid/DashboardEmpty.tsx

```diff
@@ -32,8 +37,9 @@ export const DashboardEmpty = ({ dashboard, canCreate }: Props) => {
             aria-label="Add new panel"
             onClick={() => {
               reportInteraction('Create new panel');
-              const id = onCreateNewPanel(dashboard);
+              const id = onCreateNewPanel(dashboard, initialDatasource);
```

Comment:
we would need the same functionality here: https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/components/AddPanelButton/AddPanelMenu.tsx#L30